### PR TITLE
Fix spatial utility for regrid

### LINKED ISSUE #1718: regrid nearest with mask fails for input grid extent less than target grid extent
When running land-sea-aware regridding there is an issue where the input grid-extent is smaller than the target grid-extent ( there are output points where no mapping to input points)

Command run:
```
improver regrid --regrid-mode nearest-with-mask input.nc output_grid.nc input_landmask.nc --output output.nc
```
input_landmask.nc and input.nc have grid extents **smaller** than that of output_grid.nc

The error:
```
improver/utilities/spatial.py", line 458, in maximum_within_vicinity
    unmasked_cube_data[cube.data.mask] = -np.inf
OverflowError: cannot convert float infinity to integer
```
The `-np.inf` does not convert to an integer, replacing this with a regular number (eg. netcdf fill-value for integers) would solve the issue.

Acceptance criteria:
 * Agreed approach to handling initialisation of unmasked_cube_data 
 * Code changes so CLI runs successfully independent of size of relevant extents of initial and target grids
 * All acceptance and unit tests pass with code-changes implemented.

### PR DESCRIPTION
**Pull request by anja-bom**
Resolves #1718

Have  replaced uses of -np.inf in maximum_within_vicinity function inside spatial.py which is called by land-sea aware regridding. This fixes the error of converting -np.inf to an int  in the BOM regridding example mentioned in issue #1718. 

Tried "extrapolate"  extrapolation method, found this was not being passed to AdjustLandSeaPoints - have fixed this. Please let me know if this is ok or not.

I tested the outputs for regridding with plain "nearest" and land-sea-aware version and got difference all along the coastline as expected. 

Unit and acceptance tests were run and succeeded.

Testing:
 - [x] Ran tests and they passed OK

CLA
 - [x] If a new developer, signed up to CLA

### CODE REVIEW

**Review by tjtg — COMMENTED** [review]
Changes here make sense and look like they will fix the bugs that have been encountered. It would be useful to add tests for this case where the input domain doesn't have full coverage of output domain to avoid similar issues occurring again in the future.

File: improver/utilities/spatial.py
Comment (anja-bom) [reply improver/utilities/spatial.py:468]: done

**Comment by anja-bom** [comment]
> Changes here make sense and look like they will fix the bugs that have been encountered. It would be useful to add tests for this case where the input domain doesn't have full coverage of output domain to avoid similar issues occurring again in the future.

have created issue to address this after the bugs are fixed through this merge

see #1737

File: improver/utilities/spatial.py
Comment (dmentipl) [inline improver/utilities/spatial.py:476]: I'm wondering, should the right hand side be `cube_fill_value` *not* `-cube_fill_value`? Or should the default fill value of `-np.inf` be `np.inf`? (See also line 481 below.)

File: CONTRIBUTING.md
Comment (dmentipl) [inline CONTRIBUTING.md:72]: I think this list is supposed to be in last-name-alphabetical order.

**Review by dmentipl — COMMENTED** [review]
Thanks @anja-bom, these changes look good. I've made a couple of minor comments below.

File: improver/utilities/spatial.py
Comment (fionaRust) [reply improver/utilities/spatial.py:476]: I think this needs to be a large negative value. This is because we are trying to find the maximum in an area, and we don't want any masked values to be taken into account. If we replace the missing data with a large negative number before taking the maximum the masked values will never be the maximum. These missing values never make it into the output cube, they are just manipulated in the calculation.

**Review by fionaRust — COMMENTED** [review]
I can see why both of these changes are needed and agree they are sensible fixes for the bugs you encountered. I am worried about the lack of unit tests for this PR, although I know you've created a separate issue to cover this. Why have you decided to separate out the unit tests into a separate issue?

File: improver/utilities/spatial.py
Comment (anja-bom) [reply improver/utilities/spatial.py:476]: Many tests fail if this is positive. Needs to be negative.

File: improver/utilities/spatial.py
Comment (anja-bom) [reply improver/utilities/spatial.py:470]: Done. In instances such as these it would be great if you can add a "suggested change". (saves me work)

File: CONTRIBUTING.md
Comment (anja-bom) [reply CONTRIBUTING.md:72]: Fixed

**Comment by anja-bom** [comment]
> I can see why both of these changes are needed and agree they are sensible fixes for the bugs you encountered. I am worried about the lack of unit tests for this PR, although I know you've created a separate issue to cover this. Why have you decided to separate out the unit tests into a separate issue?

Hi Fiona, this PR is holding up other work so we want to merge this quickly then start the unit tests.

File: improver/utilities/spatial.py
Comment (dmentipl) [reply improver/utilities/spatial.py:476]: In that case, the default value should be positive infinity, as you are taking the negative on line 476. Thus I think line 470 should be

```python
cube_fill_value = np.inf
```

**Review by tjtg — APPROVED** [review]

**Pull request merged**
